pennylane/templates/subroutines/uccsd_unitary.py

```diff
@@ -0,0 +1,230 @@
+           ``len(ph)+len(pphh)``
+
+
+        An example of how to use this template is shown below:
+
+        .. code-block:: python
+
+            import pennylane as qml
+            from pennylane.templates import UCCSD
+
+            # Number of active electrons
+            n_active_electrons = 2
+
+            # Number of active spin-orbitals
+            n_active_orbitals = 4
+
+            # Set the ``ref_state`` array to encode the HF state
+            ref_state = qml.qchem.hf_state(n_active_electrons, n_active_orbitals)
+
+            # Spin-projection selection rule to generate the excitations
+            DSz = 0
+
+            # Generate the particle-hole configurations
+            ph, pphh = qml.qchem.sd_excitations(n_active_electrons, n_active_orbitals, DS=DSz)
+
+            dev = qml.device('default.qubit', wires=n_active_orbitals)
+
+            @qml.qnode(dev)
+            def circuit(weights, wires, ph=ph, pphh=pphh, init_state=ref_state):
+                UCCSD(weights, wires, ph=ph, pphh=pphh, init_state=ref_state)
+                return qml.expval(qml.PauliZ(0))
+
+            params = np.random.normal(0, np.pi, len(ph) + len(pphh))
+            print(circuit(params, wires, ph=ph, pphh=pphh, init_state=ref_state))
+
+    """
+
+    ##############
+    # Input checks
+
+    check_no_variable(wires, msg="'wires' cannot be differentiable")
+
+    wires = check_wires(wires)
+
+    if (not ph) and (not pphh):
+        raise ValueError(
+            "'ph' and 'pphh' lists can not be both empty; got ph={}, pphh={}".format(ph, pphh)
+        )
+
+    check_type(ph, [list], msg="'ph' must be a list; got {}".format(ph))
+    for i_ph in ph:
+        check_type(i_ph, [list], msg="Each element of 'ph' must be a list; got {}".format(i_ph))
+        for i in i_ph:
+            check_type(
+                i, [int], msg="Each element of 'ph' must be a list of integers; got {}".format(i_ph)
+            )
+
+    check_type(pphh, [list], msg="'pphh' must be a list; got {}".format(pphh))
+    for i_pphh in pphh:
+        check_type(
+            i_pphh, [list], msg="Each element of 'pphh' must be a list; got {}".format(i_pphh)
+        )
+        for i in i_pphh:
+            check_type(
+                i,
+                [int],
+                msg="Each element of 'pphh' must be a list of integers; got {}".format(i_pphh),
+            )
+
+    check_type(
+        init_state,
+        [np.ndarray],
+        msg="'init_state' must be a Numpy array; got {}".format(init_state),
+    )
+    for i in init_state:
+        check_type(
+            i,
+            [int, np.int64],
+            msg="Elements of 'init_state' must be integers; got {}".format(init_state),
+        )
+
+    expected_shape = (2,)
+    for i_ph in ph:
+        check_shape(
+            i_ph,
+            expected_shape,
+            msg="Elements of 'ph' must be of shape {}; got {}".format(
+                expected_shape, get_shape(i_ph)
+            ),
```

Comment:
>             ),

Is the comma needed here (and few other places below)?

Reply:
I noticed that black is adding the comma after formatting the file.